frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/add-capacity.spec.ts

```diff
@@ -0,0 +1,174 @@
+      pods: null,
+      formattedOSDTree: null,
+      osdIDs: null,
+    };
+
+    cy.exec('oc get storagecluster ocs-storagecluster -n openshift-storage -o json')
+      .then((res) => {
+        const storageCluster = JSON.parse(res.stdout);
+        _.set(initialState, 'storageCluster', storageCluster);
+
+        return cy.exec(
+          'oc get cephCluster ocs-storagecluster-cephcluster -n openshift-storage -o json',
+        );
+      })
+      .then((res) => {
+        const cephCluster = JSON.parse(res.stdout);
+        _.set(initialState, 'cephCluster', cephCluster);
+
+        cy.log('Check if ceph cluster is healthy before expansion');
+        expect(cephCluster.status.ceph.health).not.toBe(CLUSTER_STATUS.HEALTH_ERROR);
+
+        return cy.exec(
+          `oc -n openshift-storage rsh $(oc get po -n openshift-storage | grep ceph-operator | awk '{print$1}') ceph --conf=/var/lib/rook/openshift-storage/openshift-storage.config osd tree --format=json`,
+          { timeout: 120000 },
+        );
+      })
+      .then((res) => {
+        const osdTree = JSON.parse(res.stdout);
+        _.set(initialState, 'osdTree', osdTree);
+
+        const formattedOSDTree = createOSDTreeMap(osdTree.nodes);
+        _.set(initialState, 'formattedOSDTree', formattedOSDTree);
+
+        const osdIDs = getIds(osdTree.nodes, 'osd');
+        _.set(initialState, 'osdIDs', osdIDs);
+
+        return cy.exec('oc get po -n openshift-storage -o json');
+      })
+      .then((res) => {
+        const pods = JSON.parse(res.stdout);
+        _.set(initialState, 'pods', pods);
+
+        cy.clickNavLink(['Operators', 'Installed Operators']);
+        cy.byLegacyTestID('item-filter').type('ocs-operator');
+        cy.byTestOperatorRow('OpenShift Container Storage').click();
+        cy.byLegacyTestID('horizontal-link-Storage Cluster').click();
+        cy.byLegacyTestID('kebab-button').click();
+        cy.byTestActionID('Add Capacity').click();
+        modal.shouldBeOpened();
+
+        const initialCapcity =
+          SIZE_MAP[
+            initialState.storageCluster?.spec?.storageDeviceSets?.[0]?.dataPVCTemplate?.spec
+              ?.resources?.requests?.storage
+          ];
+        cy.byLegacyTestID('requestSize').should('have.value', String(initialCapcity));
+        cy.byTestID('provisioned-capacity').contains(
+          `${String((initialCapcity * 3).toFixed(2))} TiB`,
+        );
+        cy.byTestID('add-cap-sc-dropdown', { timeout: 10000 }).should('be.visible');
+        modal.submit();
+        modal.shouldBeClosed();
+
+        // Wait for the storage cluster to reach Ready
+        // Storage Cluster CR flickers so wait for 10 seconds
+        // Disablng until ocs-operator fixes above issue
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(10000);
+        cy.byTestOperandLink('ocs-storagecluster').click();
+        cy.byTestID('resource-status').contains('Ready', { timeout: 900000 });
```

Comment:
There are reasons why this against best practices.   
https://docs.cypress.io/guides/references/best-practices.html#Unnecessary-Waiting
https://medium.com/better-programming/dont-use-wait-in-your-cypress-tests-cbe14746da71

Reply:
I agree these are against the best practice. But issues in the operator ( as mentioned in the comment ) force us to use these hacks. :/